src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -591,6 +591,29 @@ QString QgsPostgresUtils::whereClause( QgsFeatureId featureId, const QgsFields &
 
 QString QgsPostgresUtils::whereClause( const QgsFeatureIds &featureIds, const QgsFields &fields, QgsPostgresConn *conn, QgsPostgresPrimaryKeyType pkType, const QList<int> &pkAttrs, const std::shared_ptr<QgsPostgresSharedData> &sharedData )
 {
+  auto lookupKeyWhereClause = [ = ]( bool pkIsString )
+  {
+    if ( featureIds.isEmpty() )
+      return QString();
+
+    //simple primary key, so prefer to use an "IN (...)" query. These are much faster then multiple chained ...OR... clauses
+    QString delim;
+    QString expr = QStringLiteral( "%1 IN (" ).arg( QgsPostgresConn::quotedIdentifier( fields.at( pkAttrs[0] ).name() ) );
+
+    for ( const QgsFeatureId featureId : std::as_const( featureIds ) )
+    {
+      const QVariantList pkVals = sharedData->lookupKey( featureId );
+      if ( !pkVals.isEmpty() )
+      {
+        expr += delim + ( pkIsString ? QgsPostgresConn::quotedValue( pkVals.at( 0 ).toString() ) : pkVals.at( 0 ).toString() );
```

Comment:
I'm not sure, but at first glance, the called function
https://github.com/qgis/QGIS/blob/6a9133c74773ce2c76e16fa86cd337d443c356a3/src/providers/postgres/qgspostgresconn.cpp#L1218
already contains all the necessary logic to use the quotes correctly. Are you sure you need to duplicate it here?

Reply:
You right. I fixed it!